skeletonkey/instantiate.py

```diff
@@ -24,143 +28,151 @@ def import_target(class_string: str) -> Type[Any]:
     return obj
 
 
-def instantiate_partial(config, target_keyword=TARGET_KEYWORD,_instantiate_recursive=True, **kwargs) -> Any:
-    """
-    Instantiate a functools.partial object using a Config object. The Config
-    object should contain the key "_target_" to specify the class to instantiate.
-
-    You need to call the partial object with the remaining parameters to
-    fully instantiate.
+def instantiate(*configs:Iterable[Config], 
+                instance_keyword:str=INSTANCE_KEYWORD,
+                partial_keyword:str=PARTIAL_KEYWORD,
+                fetch_keyword:str=FETCH_KEYWORD,
+                _instantiate_recursive:bool=True,
+                **kwargs) -> Any:
 
-    example usgage can be found in example/partial_instantiate_example/
+    """
+    Instantiate a class object using a Config object.
+    The Config object should contain the key "_target_" to
+    specify the class to instantiate.
 
     Args:
-        config (Config): A Config object containing the key "_target_"
-            to specify the class, along with any additional keyword
-            arguments for the class.
+        *configs (Config): A Config object containing the class to instantiate. Multiple Config objects can be provided.
+        instance_keyword (str, optional): The keyword to use to specify a full class instantiation. Defaults to "_instance_".
+        partial_keyword (str, optional): The keyword to use to specify a partial class instantiation. Defaults to "_partial_".
+        fetch_keyword (str, optional): The keyword to use to specify a fetch class instantiation. Defaults to "_fetch_".
+        _instantiate_recursive (bool, optional): Whether to recursively instantiate subconfigs. Defaults to True.
+        **kwargs: Additional keyword arguments to pass to the class constructor
 
     Returns:
-        Any: An instance of functools.partial of the specified class.
+        Any: The instantiated class object (or list of objects if multiple configs are provided)
+
+    Raises:
+        ValueError: If no valid instantiation keyword is found in the config.
+        TypeError: If any required parameters are missing from the instacne config
+        ValueError: If a config instantiated with the _fetch_ keyword has additional arguments.
     """
+    
+    if len(configs) == 1:
+        return _instantiate_single(configs[0], instance_keyword, partial_keyword, fetch_keyword, _instantiate_recursive, **kwargs)
+    
+    else:
+        return [_instantiate_single(config, instance_keyword, partial_keyword, fetch_keyword, _instantiate_recursive, **kwargs)
+                for config in configs]
+            
+
+def _instantiate_single(config: Config,
```

Comment:
Not consistent styling with the rest of skeletonkey. Please do something like:

def _instantiate_single(
    config: Config,
    instance_keyword: str,
    partial_keyword: str,
          ...
    **extra_kwargs,
) -> Any:

Reply:
Fixed